app/scripts/modules/core/account/account.service.ts

```diff
@@ -182,8 +182,8 @@ export class AccountService {
         const available: string[] = intersection(all, this.cloudProviderRegistry.listRegisteredProviders());
         let result: string[];
         if (application) {
-          if (application.attributes.cloudProviders) {
-            result = application.attributes.cloudProviders.split(',');
+          if (application.attributes.cloudProviders.length) {
```

Comment:
I think the old `if` was doing a null check on `cloudProviders`, which would fail in your new `if`. But I think you've guaranteed that `application.attributes.cloudProviders` exists.